File: Backendspr/src/main/java/com/example/backendspr/models/Exercise.java
package com.example.backendspr.Models;

import lombok.*;

import jakarta.persistence.*;
import java.io.Serializable;


@Entity
@Getter
@Setter
@ToString
@EqualsAndHashCode
@AllArgsConstructor
@NoArgsConstructor
@Table( name = "Exercise")
public class Exercise implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nom;
    @Enumerated(EnumType.STRING)
    private Eexercice type;
    private String description;
    private String image;


}

File: FrontendAng/src/app/Core/Exercise.ts
export enum ExerciseType {
    CARDIO = "CARDIO",
    CROSS = "CROSS",
    GYMNAS = "GYMNAS",
    MEDITATION = "MEDITATION",
  }
  
  
  
  export class Exercise {
    id!: string;
    nom!: string;
    type!: ExerciseType;
    description!: string;
    image!: string;
  }